lib/temporalio/worker/runner.rb

```diff
@@ -0,0 +1,58 @@
+module Temporalio
+  class Worker
+    class Runner
+      def initialize(*workers)
+        if workers.empty?
+          raise ArgumentError, 'Must be initialized with at least one worker'
+        end
+
+        @workers = workers
+        @mutex = Mutex.new
+        @started = false
+        @shutdown = false
+      end
+
+      def run(&block)
+        @thread = Thread.current
+        @started = true
+        workers.each { |worker| worker.start(self) }
```

Comment:
So if a worker fatally errors, what happens here?

Reply:
Any unhandled worker error will initiate a shutdown at https://github.com/temporalio/sdk-ruby/pull/99/files#diff-913ed58d77d1d4e414bfe371e4670b2f60cd30c4c94e0ed3dd39ad7e7fa7ad3dR70. This in turn will complete the shutdown and then raise from a thread in which the `Runner` was called

Reply:
I see, when `start` is discouraged from use, that makes sense.